src/Workspaces/Core/Portable/CodeRefactorings/CodeRefactoringContext.cs

```diff
@@ -48,14 +62,16 @@ public struct CodeRefactoringContext
         /// Add supplied <paramref name="action"/> to the list of refactorings that will be offered to the user.
         /// </summary>
         /// <param name="action">The <see cref="CodeAction"/> that will be invoked to apply the refactoring.</param>
-        public void RegisterRefactoring(CodeAction action)
+        public void RegisterRefactoring(CodeAction action) => RegisterRefactoring(action, null);
```

Comment:
so this.Span is the selection or caret position right? which currently what engine uses for applicable span
http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Wpf/Suggestions/SuggestedActionsSource.cs,672

so here. I would do something like RegisterRefactoring(action, Span) which will retain same behavior as before. and mark this API [Obsolete] and make the other one public.

give Span here let you remove ? and remove any special case on engine side.

(action, span) =>

Reply:
I'm not sure I follow the first question the `TextSpan` here is the span where the CodeAction (Refactoring) should be applicable. If it doesn't get populated here then a `default(TextSpan)` is used for the Action's `applicableToSpan` which might have small ordering implications but nothing too drastic in comparison to what happens today (static ordering by attribute).

As for the ordering engine, changes there will come soon. 
- Improve the comparer for `applicableToSpan` so it uses selection info properly 
- Potentially remove the difference in handling refactorings and codeFixes (as you're saying).

Reply:
I understand the question about the span now :). It's being discussed below.